src/scanner/scanNFTData.js

```diff
@@ -0,0 +1,74 @@
+const { processNftEventData, processNftExtrinsicData, updateProcessedBlockInDB, updateFinalizedBlock } = require('./utils');
+const { LastBlockScan  } = require('../mongo/models');
+const { Api } = require('@cennznet/api');
+require("dotenv").config();
+const logger = require('../logger');
+const mongoose = require('mongoose');
+const BigNumber = require('bignumber.js');
+
+async function processDataAtBlockHash(api, blockHash, processingOldBlock) {
+    const events = await api.query.system.events.at(blockHash);
+    await Promise.all(
+        events.map(async ({event}) => {
+            const {section, method, data} = event;
+            if (section === 'nft') {
+                const dataFetched = data.toHuman();
+                await processNftEventData(dataFetched, method, api, processingOldBlock ? blockHash : null);
+            }
+        })
+    );
+    const block = await api.rpc.chain.getBlock(blockHash);
```

Comment:
if block is not stored in state this could be null/error

Reply:
I will have a conditional check to see if block exist then continue else log...
Even in uncover backend we are using rpc.chain.getBlock to get block at a particular hash...